Source/Urho3D/Replica/ClientReplica.cpp

```diff
@@ -158,41 +142,41 @@ bool ClientReplica::ProcessMessage(NetworkMessageId messageId, MemoryBuffer& mes
     {
     case MSG_SCENE_CLOCK:
     {
-        const auto msg = ReadNetworkMessage<MsgSceneClock>(messageData);
-        connection_->OnMessageReceived(messageId, msg);
+        const auto msg = ReadSerializedMessage<MsgSceneClock>(messageData);
+        connection_->LogReceivedMessage(messageId, msg);
 
         ProcessSceneClock(msg);
         return true;
     }
 
     case MSG_REMOVE_OBJECTS:
     {
-        connection_->OnMessageReceived(messageId, messageData);
-
         ProcessRemoveObjects(messageData);
         return true;
     }
 
     case MSG_ADD_OBJECTS:
+    case MSG_ADD_OBJECTS_INCOMPLETE:
     {
-        connection_->OnMessageReceived(messageId, messageData);
+        LargeMessageReader reader{*connection_, MSG_ADD_OBJECTS_INCOMPLETE, MSG_ADD_OBJECTS};
+        reader.OnMessage(messageId, messageData, //
+            [this](MemoryBuffer& fullMessageData) { ProcessAddObjects(fullMessageData); });
```

Comment:
Ideally stitching back large messages would be handled without user having to care for it. I can see issues it would introduce though. At very least we would waste one bit from the package id, which might be acceptable, but i am not sure. For now this is fine as such messages would be a rare case.